README.md
Neurophysiology software used by the Dr. Kendall F. Morris lab 
at the University of South Florida, Tampa, Florida.

Dr. Morris <kmorris5@usf.edu>

Maintainer:

Dale Shuman <dshuman@usf.edu>



brainstem/ChangeLog
2020-09-17  dshuman@usf.edu

	* configure.ac: Bump up to 1.1.13 for release.

2020-09-14  dshuman@usf.edu

	* brain_impl.cpp: Support for showing the control and sibling CTHs where a
	fixed set of 1 or more clusters are shown in one brainstem and the same
	cells regardless of cluster are shown in the sibling brainstem. Add command
	to generate a report showing the cluters for each CTH.
	* brainstemgl.cpp: openGl code to support the above.
	* brainstemgl.glsl: openGL shader code to support the above.

2020-08-31  dshuman@usf.edu

	* brain_impl.cpp: elimate some dead code. Looking into mapping cells in a
	cluster to ctl/stim cells regardless of cluster.

2019-11-01  dshuman@usf.edu

	* Makefile.am: move up from  g++14 to g++17

2018-12-06  dshuman@health.usf.edu

	* brainstem.desktop: fix this to point to the real locations of files.

2018-10-25  dshuman@health.usf.edu

	* brain_impl.cpp: mods to support lareflex CTHs.
	* debian/control: bump up dependency on cth_cluster to current version,
	1.3.14.
	* configure.ac: bump to version 1.1.12

2018-09-20  dshuman@health.usf.edu

	* brainstem.h: add in support for lareflex periods.
	* brain_impl.cpp: ditto.
	* configure.ac: bump to 1.1.11

2018-04-27  dshuman@health.usf.edu

	* brain_impl.cpp: add in check for flat "type" (400) and not treat it as an
	archetype.
	* configure.ac: bump up to 1.1.10.

2018-03-12  dshuman@health.usf.edu

	* brain_impl.cpp: add in check for cells at origin, which means no
	stereotaxic coordinates, so don't plot them.

2018-02-22  dshuman@health.usf.edu

	* brainstemgl.cpp: Major rewrite of how the OIT buffers are managed. I
	think cramming a float into a byte caused incorrect rendering when zooming
	in, because all the fragments were rounded down to the same value. Used the
	glslcookbook method.
	* brainstemgl.glsl: and, ditto.

2018-02-21  dshuman@health.usf.edu

	* brainstemgl.h: Mods to conform to changes to new libglm-dev package.
	Files renamed and moved.
	* stem2gl.cpp: libglm-dev ditto.

2018-02-16  dshuman@health.usf.edu

	* brainstemgl.glsl: Discover that the new sorting code sorts fragments in
	the wrong order, so things in back are in front. Fix it, then discover that
	a merge sort is even faster. I left the other sort code in the file,
	commented out.

2018-01-03  dshuman@health.usf.edu

	* brainstem.h: add in support for swallow1 cths.

2017-10-26  dshuman@health.usf.edu

	* debian/postinst.template: Move old tar file to /lab/SOURCES/old.

2017-08-31  dshuman@health.usf.edu

	* brainstem.pro: need to add in -std=c++14 explicitly for the mxe build
	environment.

2017-08-01  dshuman@health.usf.edu

	* brain_impl.cpp: fix bug where .csv files with no cths causes crash.
	* configure.ac: bump to 1.1.8.

2017-07-26  dshuman@health.usf.edu

	* configure.ac: bumpt o 1.1.7 for new release.
	* brainstem.ui: On some systems, the font is too large, overflowed the
	buttons. Adjust the gui so the window resize makes the buttons bigger, and
	also allow a bigger resize (don't know why I limited the left control
	window to be 600 pixels.)

2017-06-29  dshuman@health.usf.edu

	* brainstemgl.glsl: experiment with different sorting algorithms. Change
	from bubble sort to insert sort. A bit faster.

2017-06-27  dshuman@health.usf.edu

	* brainstemgl.cpp: move the shader code to its own file so syntax
	highlighting would work, plus, helps unclutter this file.
	* brainstemgl.glsl: moved the shader code from the above file to here. New
	file.
	* Makefile.am: Add above file to srcs list.

2017-06-16  dshuman@health.usf.edu

	* brain_impl.cpp: 
	* brainstemgl.cpp: Redo how movies are made. Let brain_impl, not brainstemgl
	decide when to grab a new frame. I was grabbing the same frame twice in
	some cases. Video seems a bit smoother now. Also add ability to make a
	movie with no spin or twinkle. Display a movie time now, not # of frames.


2017-06-07  dshuman@health.usf.edu

	* brain_impl.cpp: Force opengl framebuffer to be even W and H. I was
	telling the ffmpeg program to scale the FB after the fact, which resulted
	in a somewhat blurry image in .wmv movies.
	Also pass just file name to movie saving function, so we don't get
	ever-increasing .wmv and .avi extensions.
	* brainstemgl.cpp: Remove scaling args to the ffmpeg command line for .wmv
	files.

2017-04-21  dshuman@health.usf.edu

	* brain_impl.cpp: Add in support for archetype clustering. New .csv file
	format is supported, previous format is still supported.
	* brainstem.h: Support for archetype clustering files.

2017-03-23  dshuman@health.usf.edu

	* configure.ac: bump to 1.1.5.

2017-02-08  dshuman@health.usf.edu

	* brain_impl.cpp: Add in saving and restoring figure settings by filename
	to support creating documents. Kind of like style sheets. Add in jitter for
	.dx files.
	* configure.ac: version 1.1.4.
	* brainstemgl.cpp: Support of saving and restoring figure settings, see
	above.

2017-01-25  dshuman@health.usf.edu

	* brain_impl.cpp: don't let animate timer be 0, movies fail to save.

2017-01-24  dshuman@health.usf.edu

	* brainstemgl.cpp: Fix bug in making .wmv file, window w/h must be
	multiples of 2.

2017-01-11  dshuman@health.usf.edu

	* brain_impl.cpp: Fix a crash problem with bad .dx files. Now just refuse
	to load them.
	* configure.ac: bump to 1.1.3.

2017-01-05  dshuman@health.usf.edu

	* debian/control: Remove dependency and qt5-style-plugins. If this is
	present, the GUI is horrible. Prevously, you needed it with Qt 5.6 because
	without it, the GUI was horrible. Qt 5.7 showed up recently, and now things
	break differently.
	* configure.ac: bump to 1.1.2 for new release.

2016-12-14  dshuman@health.usf.edu

	* Version 1.1.1 release.
	* Makefile.am: add rule so Makefile.qt, which is only used by qtcreator
	while debugging, picks up VERSION var.

2016-12-13  dshuman@health.usf.edu

	* debian/control: Add ffmpeg as a dependency.
	* configure.ac: bump to 1.1.1.

2016-12-09  dshuman@health.usf.edu

	* configure.ac: release of v 1.1.0


brainstem/atlasnames.c
/* Copyright 2005-2020 Kendall F. Morris

This file is part of the USF Brainstem Data Visualization suite.

    The Brainstem Data Visualiation suite is free software: you can
    redistribute it and/or modify it under the terms of the GNU General Public
    License as published by the Free Software Foundation, either version 3 of
    the License, or (at your option) any later version.

    The suite is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with the suite.  If not, see <https://www.gnu.org/licenses/>.
*/


const char *structNames[] =
{
 "CENTRAL CANAL",// value = "2"
 "INF CENTRAL N",// value = "3"
 "POSTPYRAMIDAL NUCLEUS, RAPHE",// value = "5"
 "L ABDUCENS NERVE",// value = "6"
 "R ABDUCENS NERVE",// value = "7"
 "R ABDUCENS NUC",// value = "9"
 "L INF CEREBELLAR PEDUNCLE",// value = "10"
 "R INF CEREBELLAR PEDUNCLE",// value = "11"
 "L MID CEREBELLAR PEDUNCLE",// value = "12"
 "R MID CEREBELLAR PEDUNCLE",// value = "13"
 "L SUP CEREBELLAR PEDUNCLE",// value = "14"
 "R SUP CEREBELLAR PEDUNCLE",// value = "15"
 "L A-V COCHLEAR NUC",// value = "16"
 "R A-V COCHLEAR NUC",// value = "17"
 "L DOR COCHLEAR NUC",// value = "18"
 "R DOR COCHLEAR NUC",// value = "19"
 "L P-V COCHLEAR NUC",// value = "20"
 "R P-V COCHLEAR NUC",// value = "21"
 "L COCHLEAR GRAN LAYER",// value = "22"
 "R COCHLEAR GRAN LAYER",// value = "23"
 "L CUNEATE NUC",// value = "26"
 "R CUNEATE NUC",// value = "27"
 "L EX CUNEATE NUC",// value = "28"
 "R EX CUNEATE NUC",// value = "29"
 "L CUNEATE FASC",// value = "30"
 "R CUNEATE FASC",// value = "31"
 "L FACIAL NERVE",// value = "34"
 "R FACIAL NERVE",// value = "35"
 "L FACIAL NUC",// value = "36"
 "R FACIAL NUC",// value = "37"
 "L GRACILE NUC",// value = "38"
 "R GRACILE NUC",// value = "39"
 "L GRACILE FASC",// value = "40"
 "R GRACILE FASC",// value = "41"
 "L HYPOGLOSSAL NUC",// value = "42"
 "R HYPOGLOSSAL NUC",// value = "43"
 "L INFRATRIGEMINAL NUC",// value = "44"
 "L INFRATRIGEMINAL NUC",// value = "45"
 "L INF OLIVE COMPLEX",// value = "46"
 "R INF OLIVE COMPLEX",// value = "47"
 "L KOLLIKER-FUSE NUC",// value = "50"
 "R KOLLIKER-FUSE NUC",// value = "51"
 "L NUC-LAT LEMNISCUS (VEN DIV)",// value = "54"
 "R NUC-LAT LEMNISCUS (VEN DIV)",// value = "55"
 "L LAT RETICULAR NUC",// value = "56"
 "R LAT RETICULAR NUC",// value = "57"
 "L LAT RETICULAR NUC",// value = "58"
 "R LAT RETICULAR NUC",// value = "59"
 "L MARGINAL NUC, SUPERIOR CEREBELLAR PEDUNCLE",// value = "60"
 "R MARGINAL NUC, SUPERIOR CEREBELLAR PEDUNCLE",// value = "61"
 "L MED LONGITUDINAL BUNDLE",// value = "62"
 "R MED LONGITUDINAL BUNDLE",// value = "63"
 "L NUC AMBIGUUS",// value = "64"
 "R NUC AMBIGUUS",// value = "65"
 "L NUC INCERTUS",// value = "66"
 "R NUC INCERTUS",// value = "67"
 "L NUC INTERCALATUS",// value = "68"
 "R NUC INTERCALATUS",// value = "69"
 "L LOCUS COERULEUS",// value = "70"
 "R LOCUS COERULEUS",// value = "71"
 "L NUC PRAEPOSITUS HYPOGLOSSI",// value = "72"
 "R NUC PRAEPOSITUS HYPOGLOSSI",// value = "73"
 "L NUC RETROAMBIGUALIS",// value = "74"
 "R NUC RETROAMBIGUALIS",// value = "75"
 "L NUC TRACTUS SOLITARIUS",// value = "76"
 "R NUC TRACTUS SOLITARIUS",// value = "77"
 "L PONTINE GRAY (D-L DIV)",// value = "78"
 "R PONTINE GRAY (D-L DIV)",// value = "79"
 "L RETROFACIAL NUC",// value = "84"
 "R RETROFACIAL NUC",// value = "85"
 "L SUP OLIVE COMPLEX (LAT HUC)",// value = "86"
 "R SUP OLIVE COMPLEX (LAT NUC)",// value = "87"
 "L SUP OLIVE COMPLEX (MED NUC)",// value = "88"
 "R SUP OLIVE COMPLEX (MED NUC)",// value = "89"
 "L SUP OLIVE COMPLEX (PREOLIVARY)",// value = "90"
 "R SUP OLIVE COMPLEX (PREOLIVARY)",// value = "91"
 "L STATOACOUSTIC NERVE",// value = "94"
 "R STATOACOUSTIC NERVE",// value = "95"
 "L GIGANTOCELLULAR TEG FIELD",// value = "96"
 "R GIGANTOCELLULAR TEG FIELD",// value = "97"
 "L LAT TEGMENTAL FIELD",// value = "98"
 "R LAT TEGMENTAL FIELD",// value = "99"
 "L MAGNOCELLULAR TEGMENTAL FIELD",// value = "100"
 "R MAGNOCELLULAR TEGMENTAL FIELD",// value = "101"
 "L DOR TEGMENTAL NUC (PERICEN)",// value = "104"
 "R DOR TEGMENTAL NUC (PERICEN)",// value = "105"
 "L TEGMENTAL RETICULAR NUC",// value = "108"
 "R TEGMENTAL RETICULAR NUC",// value = "109"
 "L TRIGEMINAL NERVE (V)",// value = "110"
 "R TRIGEMINAL NERVE (V)",// value = "111"
 "L SENSORY NUC OF V",// value = "112"
 "R SENSORY NUC OF V",// value = "113"
 "L SPINAL TR OF V",// value = "114"
 "R SPINAL TR OF V",// value = "115"
 "L MOTOR TR OF V",// value = "116"
 "R MOTOR TR OF V",// value = "117"
 "L CAUDAL NUC, SPINAL TR OF V",// value = "118"
 "R CAUDAL NUC, SPINAL TR OF V",// value = "119"
 "L INTERPOLAR NUC, SPINAL TR OF V",// value = "120"
 "R INTERPOLAR NUC, SPINAL TR OF V",// value = "121"
 "L ORAL NUC, SPINAL TR OF V",// value = "122"
 "R ORAL NUC, SPINAL TR OF V",// value = "123"
 "L SOLITARY TRACT",// value = "126"
 "R SOLITARY TRACT",// value = "127"
 "L STRIA ACUSTICA",// value = "128"
 "R STRIA ACUSTICA",// value = "129"
 "L NUC, TRAPEZOID BODY",// value = "130"
 "R NUC, TRAPEZOID BODY",// value = "131"
 "L TRAPEZOID BODY",// value = "132"
 "R TRAPEZOID BODY",// value = "133"
 "L INF VESTIBULAR NUC",// value = "134"
 "R INF VESTIBULAR NUC",// value = "135"
 "L MED VESTIBULAR NUC",// value = "136"
 "R MED VESTIBULAR NUC",// value = "137"
 "L LAT VESTIBULAR NUC (DOR DIV)",// value = "138"
 "R LAT VESTIBULAR NUC (DOR DIV)",// value = "139"
 "R LAT VESTIBULAR NUC (VEN DIV)",// value = "141"
 "L SUP VESTIBULAR NUC (MED DIV)",// value = "142"
 "R SUP VESTIBULAR NUC (MED DIV)",// value = "143"
 "L SUP VESTIBULAR NUC (LAT DIV)",// value = "144"
 "R SUP VESTIBULAR NUC (LAT DIV)",// value = "145"
 "L DOR MOTOR NUC, VAGUS",// value = "146"
 "R DOR MOTOR NUC, VAGUS",//  value = "147"
};

int totNames=sizeof(structNames)/sizeof(char*);

 
 
